pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,957 +71,765 @@
-            # https://github.com/PyTorchLightning/pytorch-lightning/pull/1572/files#r420279383
-            os.environ["HOROVOD_FUSION_THRESHOLD"] = str(0)
-            # https://docs.nvidia.com/cuda/cublas/index.html#cublasApi_reproducibility
-            os.environ["CUBLAS_WORKSPACE_CONFIG"] = ":4096:8"
-
-    def select_accelerator_type(self) -> None:
-        if self.distributed_backend == "auto":
-            if self.has_tpu:
-                self._accelerator_type = _AcceleratorType.TPU
-            elif self.has_ipu:
-                self._accelerator_type = _AcceleratorType.IPU
-            elif self.has_gpu:
-                self._accelerator_type = _AcceleratorType.GPU
-            else:
-                self._set_devices_to_cpu_num_processes()
-                self._accelerator_type = _AcceleratorType.CPU
-        elif self.distributed_backend == _AcceleratorType.TPU:
-            if not self.has_tpu:
-                msg = "TPUs are not available" if not _TPU_AVAILABLE else "you didn't pass `tpu_cores` to `Trainer`"
-                raise MisconfigurationException(f"You passed `accelerator='tpu'`, but {msg}.")
-            self._accelerator_type = _AcceleratorType.TPU
-        elif self.distributed_backend == _AcceleratorType.IPU:
-            if not self.has_ipu:
-                msg = "IPUs are not available" if not _IPU_AVAILABLE else "you didn't pass `ipus` to `Trainer`"
-                raise MisconfigurationException(f"You passed `accelerator='ipu'`, but {msg}.")
-            self._accelerator_type = _AcceleratorType.IPU
-        elif self.distributed_backend == _AcceleratorType.GPU:
-            if not self.has_gpu:
-                msg = "you didn't pass `gpus` to `Trainer`" if torch.cuda.is_available() else "GPUs are not available"
-                raise MisconfigurationException(f"You passed `accelerator='gpu'`, but {msg}.")
-            self._accelerator_type = _AcceleratorType.GPU
-        elif self.distributed_backend == _AcceleratorType.CPU:
-            self._set_devices_to_cpu_num_processes()
-            self._accelerator_type = _AcceleratorType.CPU
-
-        if self.distributed_backend in self.accelerator_types:
-            self.distributed_backend = None
-
-    def _validate_accelerator_and_devices(self) -> None:
-        if self.distributed_backend not in self.accelerator_types and self.devices is not None:
-            raise MisconfigurationException(
-                f"You passed `devices={self.devices}` but haven't specified"
-                " `accelerator=('auto'|'tpu'|'gpu'|'ipu'|'cpu')` for the devices mapping,"
-                f" got `accelerator={self.distributed_backend!r}`."
-            )
+        # --Parsing_flags------------------------------------------------------
+        # Get registered strategies, existing accelerators and precision plugins
+        self._existing_strategies_str = StrategyRegistry.available_strategies()
+        self._existing_accelerator_type = ["tpu", "ipu", "gpu", "cpu"]
```

Comment:
```suggestion
        self._existing_accelerator_type = ("tpu", "ipu", "gpu", "cpu")
```

don't use mutable types unless necessary :) Avoids accidental changes